paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt

```diff
@@ -104,36 +113,57 @@ internal class DefaultPaymentSheetLoader @Inject constructor(
         customerConfig: PaymentSheet.CustomerConfiguration?,
         config: PaymentSheet.Configuration?,
         isGooglePayReady: Boolean,
-        isLinkReady: Boolean
     ): PaymentSheetLoader.Result {
         val prefsRepository = prefsRepositoryFactory(customerConfig)
 
-        val paymentMethods = if (customerConfig != null) {
-            retrieveCustomerPaymentMethods(
-                stripeIntent,
-                config,
-                customerConfig
-            )
-        } else {
-            emptyList()
-        }
+        val isLinkAvailable = stripeIntent.paymentMethodTypes.contains(Link.code) &&
+            stripeIntent.linkFundingSources.intersect(supportedFundingSources).isNotEmpty()
 
-        val savedSelection = retrieveSavedPaymentSelection(
-            prefsRepository,
-            isGooglePayReady,
-            isLinkReady,
-            paymentMethods
-        )
+        val paymentMethods: Deferred<List<PaymentMethod>>
+        val savedSelection: Deferred<SavedSelection>
+        val linkState: Deferred<LinkState?>
+
+        coroutineScope {
+            paymentMethods = async {
+                if (customerConfig != null) {
```

Comment:
This doesn’t feel super elegant. Any ideas for how to improve it?

Reply:
What don't you like about it?